File: Easy/average-salary-excluding-the-minimum-and-maximum-salary.py
class Solution:     # 32ms
    def average(self, salary: List[int]) -> float:
        runSum = 0
        extreme = [salary[0]] * 2
        for i in range(len(salary)):
            runSum += salary[i]
            extreme[0] = max(salary[i], extreme[0])
            extreme[1] = min(salary[i], extreme[1])
        return (runSum - extreme[0] - extreme[1]) / (len(salary) - 2)

File: Easy/average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {        // 0ms
    public double average(int[] salary) {
        double[] run = {0, salary[0], salary[0]};
        for(int i = 0; i < salary.length; i++){
            run[0] += salary[i];
            run[1] = Math.min(run[1], salary[i]);
            run[2] = Math.max(run[2], salary[i]);
        }
        return (run[0] - run[1] - run[2]) / (salary.length - 2);
    }
}